### PR TITLE
optimization with cython

### PR DESCRIPTION
**Pull request by chlozhen**
- Added opt_cython folder with the entire repo copied. Only changed/compiled agent, location, and transition files with cython. 
- Added optimization_time.py to compare python vs cython method --> can time other optimization methods in the future
- Cleaned up figures - save them in basic_sim and vaccine_sim instead so that it does not interfere with timing the program

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 *.pyc
 *.xml
 *.iml
+*.png
```
Comment (jw7383) [inline .gitignore:5]: do we want to ignore .c and .so files as well? do we need those to be pushed to the repo?

**Review by jw7383 — COMMENTED** [review]
looks great! do you think you can add to the README how to run optimization_time.py, and also a very quick summary of our first optimization technique with cython?

File: .gitignore (hunk 1 above)
Comment (chlozhen) [reply .gitignore:5]: If we don't push them, whoever downloads needs to compile it themselves before running optimization_time.py - so i can add directions in the readme for that. Or we can leave them and they won't have to do that.

Comment (jw7383) [reply .gitignore:5]: we can leave it then

**Review by jw7383 — APPROVED** [review]
lgtm! thanks chloe!

**Pull request merged**